PusherClient/Pusher.cs

```diff
@@ -100,24 +101,14 @@ public Pusher(string applicationKey, PusherOptions options = null)
 
         #region Public Methods
 
+        [MethodImpl(MethodImplOptions.Synchronized)]

```

Comment:
What is the advantage of using `lock()`?

Feel free to look at it more thoroughly later. Essentially we want the Connection object to be a singleton, and any subsequent calls to Connect should be a noop.